pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationProvider.java

```diff
@@ -60,22 +59,14 @@
      */
     String authenticate(AuthenticationDataSource authData) throws AuthenticationException;
```

Comment:
Can you add a default here that throws an exception? This method is only useful in onestage auth. The comment should say that, and by default (if you're not doing onestage or if you're providing your own state implemenation for onestage) it should throw an exception.